src/central_agentic_agent.py

```diff
@@ -202,6 +212,28 @@ def get_bgt_info(self, address: str) -> str:
         except Exception as e:
             logging.error(f"Failed to get BGT info: {e}")
             return "No information found"
+        
+    def get_policy_info(self, melding: str) -> str:
+        """
+        Retrieve BGT (Basisregistratie Grootschalige Topografie) information based on the provided address.
+
+        Args:
+            address (str): The address in the format 'STRAATNAAM, HUISNUMMER, POSTCODE'.
+
+        Returns:
+            str: BGT information or 'No information found' if unsuccessful.
+        """
+        logging.info(f"Retrieving policy info for melding: {melding}")
+        try:
+            melding = self.melding_attributes['MELDING']
+            policy_retriever_info_tool = PolicyRetrieverAgent(melding)
```

Comment:
In my opinion this should not be initialized/instantiated every time. The main agent can just have an instance of all of these tools and the get policy info function can always use this one same instance. In this way we don't have to set up DB connections, load models, etc every time.
This will become an issue when I swap to the open source models, as e.g. the underlying prompt summarization step would require loading up a model every single time